contents/blog/a-non-coders-thoughts-on-everybody-codes-culture-part-two.md

```diff
@@ -0,0 +1,54 @@
+featuredImageType: full
+---
+
+One of PostHog’s values is that everybody codes. That doesn’t mean everyone needs to bea developer, but we do encourage everyone to practice the basics of shipping, no matter their role. Even site updates require GitHub and  some basic knowledge of coding, for example. 
+
+But, of course, you may know that already. [I wrote about it a year ago, when I joined PostHog](/blog/a-non-coders-thoughts-on-everybody-codes-culture). Back then, my coding knowledge was limited to just a smattering of HTML.
+
+Not anymore. PostHog is still unlikely to consider me a good candidate for [an engineering  role](/careers), but I’m _much_ more confident with code. GitHub is no longer foreign ground and yesterday I juggled React, Javascript and HTML in a way which _mostly_ worked! 
+
+In my previous post I wrote about the three principles that set me up for success, but now it’s time to reflect on what’s helped me succeed since. 
+
+### Having unrealistic goals
+In my last post [I said it was important to set realistic limits](blog/a-non-coders-thoughts-on-everybody-codes-culture) and that’s true, at the start.
+
+However, it’s also been important to follow less grounded ideas. One example is the ShuffleHog — an idea I had for a hackathon project which is essentially a shuffle button to help you discover new insights. I thought ShuffleHog was a fun idea, but nobody else latched on to it.
+
+If I’d stuck to realistic limits, that would be the end of it. I can’t  build a product feature on my own, can I?
+
+It turns out that I can. Late last year I made [a ShuffleHog prototype on Twitter](https://twitter.com/thehogbot), coded in [Tracery](https://www.tracery.io/). That lasted until [Marius added the ability to make front-end apps in PostHog](https://github.com/PostHog/posthog/issues/9654#issuecomment-1133222836) last week. 24 hours later, ShuffleHog now lives inside PostHog as a beta feature that can suggest random query ideas.
+
+![ShuffleHog - AKA PostHog's Best Feature](../images/blog/non-coders-thoughts/shufflehog_app.png)
+
+ShuffleHog still isn’t quite ready for other users to try yet, but who knows? Maybe I’ll keep following this dream. 
+
+### (Keep) making mistakes
+One thing that hasn’t changed over the last 12 months is the importance of making mistakes — and I should know. I make a lot of them.
+
+The reason mistakes are so important is, I think, because coding isn’t a prescriptive discipline. There’s no single correct way to code or write, so it’s more valuable to know why something _broke_ than how it ‘should’ be done. 
+
+Breaking things is a great way to learn about how to fix them.PostHog recently rebranded our plugins to apps and, as part of that, we launched a new apps library.  I broke a bunch of things while working on that project — but that's OK because I managed to fix most of them (with occasional help from Eli) and carried those lessons over to the ShuffleHog project.
```

Comment:
```suggestion
Breaking things is a great way to learn about how to fix them. PostHog recently rebranded plugins to apps and, as part of that, we launched a new apps library. I broke a bunch of things while working on that project — but that's OK because I managed to fix most of them (with occasional help from Eli) and carried those lessons over to the ShuffleHog project.
```